dashboard/client/src/components/NewEventTable.tsx

```diff
@@ -0,0 +1,335 @@
+  Object.entries(filters).forEach(([key, value]) => {
+    if (key !== "entityName" && key !== "entityId") {
+      appendToParams(params, key, value as string | string[]);
+    }
+  });
+
+  return params.toString();
+};
+
+const useStyles = makeStyles((theme) => ({
+  overflowCell: {
+    display: "block",
+    margin: "auto",
+    maxWidth: 360,
+    textOverflow: "ellipsis",
+    overflow: "hidden",
+    whiteSpace: "nowrap",
+  },
+  filterContainer: {
+    display: "flex",
+    alignItems: "center",
+  },
+  search: {
+    margin: theme.spacing(1),
+    display: "inline-block",
+    fontSize: 12,
+    lineHeight: "46px",
+    height: 56,
+  },
+
+  tableContainer: {
+    overflowX: "scroll",
+  },
+
+  helpInfo: {
+    marginLeft: theme.spacing(1),
+  },
+  message: {
+    maxWidth: 200,
+  },
+  pagination: {
+    marginTop: theme.spacing(3),
+  },
+}));
+
+const SOURCE_TYPE_OPTIONS = [
+  "COMMON",
+  "CORE_WORKER",
+  "GCS",
+  "RAYLET",
+  "JOBS",
+  "SERVE",
+  "CLUSTER_LIFECYCLE",
+  "AUTOSCALER",
+];
+
+const SEVERITY_LEVEL_OPTIONS = ["INFO", "DEBUG", "WARNING", "ERROR", "TRACING"];
+
+const COLUMNS = [
+  { label: "Severity", align: "center" },
+  { label: "Timestamp", align: "center" },
+  { label: "Source", align: "center" },
+  { label: "Custom Fields", align: "left" },
+  { label: "Message", align: "left" },
+];
+
+const useEventTable = (props: EventTableProps) => {
+  const { defaultSeverityLevels, entityName, entityId } = props;
+  const [loading, setLoading] = useState(true);
+  const { changeFilter: _changeFilter, filterFunc } = useFilter();
+  const [filters, _setFilters] = useState<Filters>({
+    sourceType: [],
+    severityLevel: defaultSeverityLevels || [],
+    entityName, // We used two fields(entityName, entityId) because we will support select entityName by dropdown and input entityId by TextField in the future.
+    entityId, // id or *
+  });
+
+  const [events, setEvents] = useState<Event[]>([]);
+
+  const [pagination, setPagination] = useState({
+    pageNo: 1, // first page is PageNo 1
+    pageSize: 10,
+  });
+
+  const changePage = (key: string, value: number) => {
+    setPagination({ ...pagination, [key]: value });
+  };
+
+  const changeFilter: typeof _changeFilter = (...params) => {
+    _changeFilter(...params);
+    setPagination({
+      ...pagination,
+      pageNo: 1,
+    });
+  };
+
+  const setFilters: typeof _setFilters = (...params) => {
+    _setFilters(...params);
+    setPagination({
+      ...pagination,
+      pageNo: 1,
+    });
+  };
+  const { pageNo } = pagination;
+
+  const params = transformFiltersToParams(filters);
+  const { data: eventsData, error, isLoading } = useEvents(params, pageNo);
+
+  useEffect(() => {
+    if (eventsData) {
+      setEvents(eventsData);
+    }
+    if (error) {
+      console.error("getEvent error: ", error);
+    }
+    setLoading(isLoading);
+  }, [eventsData, error, isLoading]);
```

Comment:
why do we have a `useEffect` here? 

why not use `eventsData`, `isLoading`, and `error` directly in the rest of the component?

Reply:
That is a good idea. We don't need this useEffect here